app/views/devise/registrations/edit.html.erb

```diff
@@ -3,51 +3,48 @@
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
   <%= devise_error_messages! %>
 
-  <div class="field">
-    <%= f.label :first_name %><br />
-    <%= f.text_field :first_name, autofocus: true %>
+  <div class="field"></br>
+    <%= f.text_field :first_name, autofocus: true, placeholder: 'first name' %>
   </div>
 
-  <div class="field">
-    <%= f.label :last_name %><br />
-    <%= f.text_field :last_name %>
+  <div class="field"></br>
+    <%= f.text_field :last_name, placeholder: 'last name' %>
   </div>
 
-  <div class="field">
-    <%= f.label :email %><br />
-    <%= f.email_field :email %>
-  </div>
+  <div class="field"></br>
+    <%= f.email_field :email, placeholder: 'e-mail address'  %>
+  </div></br>
 
   <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
     <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
   <% end %>
 
+
   <div class="field">
-    <%= f.label :password %> <i>(leave blank if you don't want to change it)</i><br />
-    <%= f.password_field :password, autocomplete: "off" %>
+    <%= f.password_field :current_password, autocomplete: "off", placeholder: 'current password' %>
+    <i> we need your current password to confirm your changes</i><br />
+  </div>
+
+  <div class="field"></br>
+    <%= f.password_field :password, autocomplete: "off", placeholder: 'new password' %>
+    <i> leave blank if you don't want to change it - </i>
     <% if @minimum_password_length %>
-      <br />
       <em><%= @minimum_password_length %> characters minimum</em>
     <% end %>
   </div>
 
-  <div class="field">
-    <%= f.label :password_confirmation %><br />
-    <%= f.password_field :password_confirmation, autocomplete: "off" %>
+  <div class="field"></br>
+    <%= f.password_field :password_confirmation, autocomplete: "off", placeholder: 'new password confirmation' %>
   </div>
 
-  <div class="field">
-    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
-    <%= f.password_field :current_password, autocomplete: "off" %>
-  </div>
 
   <div class="actions">
```

Comment:
A line break before Update button would be nice. Again, just personal preference, not necessary to incorporate.